ddtrace/opentracer/tracer.py

```diff
@@ -72,13 +73,28 @@ def __init__(self, service_name=None, config=None, scope_manager=None):
 
         self._scope_manager = scope_manager or ThreadLocalScopeManager()
 
+        # avoid having to import scope managers which may not be compatible
```

Comment:
Can you delegate this block to another function inside our utils? So you can test this behavior separately. This means having:
```python
dd_context_provider = getDDContextManager(self._scope_manager)
```

and you leave implementation details in another function. Also write a test for these providers (i.e. in `opentracer_asyncio` tox task) so we're sure they're properly mapped.